File: java/Desafio1.java
public class Desafio1 {
    
    static int a = 10;
    static int b = 20;
    
    public static void main(String args[]) {
        
        Desafio1.impressao("pre-processamento");
        Desafio1.processamento();
        Desafio1.impressao("pos-processamento");
    }
    
    private static void processamento(){
        int aux;
        
        aux = Desafio1.a;
        Desafio1.a = Desafio1.b;
        Desafio1.b = aux;
    }
    
    private static void impressao(String etapa){
        System.out.println("# " + etapa);
        System.out.println("Valor de A: " + Desafio1.a);
        System.out.println("Valor de B: " + Desafio1.b);
    }

}

File: python/Desafio1.py
class Desafio1:
    a = 10
    b = 20

    @classmethod
    def imprime(cls, etapa):
        print(f'## {etapa} ##')
        print(f'Valor de a {cls.a}')
        print(f'Valor de b {cls.b}')

    @classmethod
    def processa(cls):
        aux = cls.a
        cls.a = cls.b
        cls.b = aux
    
    @classmethod
    def main(cls):
        cls.imprime("inicio")
        cls.processa()
        cls.imprime("final")
        


Desafio1.main()